sdk/core/core/src/az_http_private.h

```diff
@@ -38,6 +38,12 @@ AZ_NODISCARD AZ_INLINE az_result _az_http_request_remove_retry_headers(_az_http_
   return AZ_OK;
 }
 
+/**
+ * @brief sets buffer and parser to its initial state
```

Comment:
nit: Docs should be full sentences and also include parameter description (and return  value where relevant)

So, capitalize and add a period.

```suggestion
 * @brief Sets buffer and parser to its initial state.
```

Also, the method is resetting an `http_response`, where did the buffer and parser come from? Are those concepts that are exposed publicly to the user?

Reply:
they came from az_http_response_init()

Reply:
That doesn't mention a parser at all. Avoid exposing internal implementation detail in docs:
https://github.com/Azure/azure-sdk-for-c/blob/257d4c77367e80e484d3c639d7cc0f6ed2aa09d4/sdk/core/core/inc/az_http.h#L163